### PR TITLE
unix: uv__fs_sendfile using copy_file_range on FreeBSD 13 and above

### PR DESCRIPTION
**Pull request by devnexen**
when possible.

### CODE REVIEW

File: src/unix/freebsd.c [hunk 1]
```diff
@@ -287,3 +287,18 @@ int uv__recvmmsg(int fd, struct uv__mmsghdr* mmsg, unsigned int vlen) {
...
+
+ssize_t
+uv__fs_copy_file_range(int fd_in,
+                       off_t* off_in,
+                       int fd_out,
+                       off_t* off_out,
+                       size_t len,
+                       unsigned int flags)
+{
+#if __FreeBSD_version >= 1300000
```
Comment (bnoordhuis) [inline src/unix/freebsd.c:299]: Shouldn't this be something like `#if __FreeBSD__ >= 13 && !defined(__DragonFly__)`, like the other guards in this files? If not, why not?

File: src/unix/fs.c [hunk 2]
```diff
@@ -1075,6 +1075,17 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
...
 
 #if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__)
+    off_t off;
+
+    off = req->off;
+    r = uv__fs_copy_file_range(in_fd, &off, out_fd, NULL, req->bufsml[0].len, 0);
+    if (r > 0) {
+        r = off - req->off;
+        req->off = r;
```
Comment (bnoordhuis) [inline src/unix/fs.c:1085]:
```suggestion
        req->off = off;
```

File: src/unix/fs.c [hunk 3]
```diff
@@ -1075,6 +1075,17 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
...
+        return r;
+    }
```
Comment (bnoordhuis) [inline src/unix/fs.c:1087]: Falls through to the sendfile() call in the `r == 0` case. Maybe that's harmless but it's still an extra system call so perhaps it shouldn't?

File: src/unix/freebsd.c (hunk 1 above)
Comment (devnexen) [reply src/unix/freebsd.c:299]: true but it s shorter at least but no pb I can change.

File: src/unix/fs.c (hunk 3 above)
Comment (devnexen) [reply src/unix/fs.c:1087]: fair enough

**Review by bnoordhuis — APPROVED** [review]

**Pull request merged**